editor/src/components/canvas/controls/select-mode/select-mode-hooks.tsx

```diff
@@ -567,7 +567,7 @@ function useSelectOrLiveModeSelectAndHover(
 
       // Skip all of this handling if 'space' is pressed or a mousemove happened in an interaction
       if (!(isSpacePressed || hasInteractionSessionWithMouseMoved)) {
-        const doubleClick = event.detail > 1 // we interpret a triple click as two double clicks, a quadruple click as three double clicks, etc
+        const doubleClick = event.type === 'mousedown' && event.detail > 0 && event.detail % 2 === 0
```

Comment:
I believe the standard click/doubleclick handling is to listen to mouse up. That is why you can "undo" a mouse down, if you move away before releasing the button. You can check this e.g. on mac is, all click events are happening on mouse up.
Or we could just use the `click` and `dblclick` events, is there a problem with those?

Reply:
Our selection logic differs between `mousedown` and `mouseup`, so if we tie this to `mouseup` only it completely breaks double click selection. If we _don't_ restrict this to just one of the two then the focusing behaviour is still incorrect (as the `mousedown` "double click" will select the element, and then the `mouseup` "double click" will immediately focus it).

If we instead switch to using `dblclick` we have to add extra logic to check if the related `mousedown` or `mouseup` events changed the selection, which is a whole new can of worms.

Reply:
I checked and even figma uses mouse down to detect double click...
This is strange for me, but I accept it :)

Reply:
I think this is because of all of the nuances around canvas editing in general - e.g. wanting to select and drag in the same action, or wanting to drag an already selected (but overlapped) element vs wanting to select the overlapping element when there is no drag.